firebase/functions/src/osm_auth.ts

```diff
@@ -10,27 +10,26 @@
 // for the 2 main functions.
 
 'use strict';
-
-const functions = require('firebase-functions');
-const cookieParser = require('cookie-parser');
-const crypto = require('crypto');
-const simpleOAuth2 = require('simple-oauth2');
-const axios = require('axios');
+import * as functions from 'firebase-functions';
+import * as crypto from 'crypto';
+import * as cookieParser from 'cookie-parser';
+import * as simpleOAuth2 from 'simple-oauth2';
+import axios from 'axios';
 
 // this redirect_uri MUST match the one set in the app on OSM OAuth, or you
 // will get a cryptic error about the server not being able to continue
 // TODO: adjust the prefix based on which deployment is done (prod/dev)
-const OAUTH_REDIRECT_URI = functions.config().osm.redirect_uri;
+const OAUTH_REDIRECT_URI = functions.config().osm?.redirect_uri;
```

Comment:
just out of curiosity: what does the `?` mean here. why did you add it?

Reply:
It checks for undefined/null values, so the above checks that `osm` is not null/undefined before trying to read `redirect_uri` from it, otherwise, it defaults to undefined instead of throwing an error.
I think this is ok if there's a check for undefined result, as the code should not run without these values set.